apps/storybook/src/DataCurve.stories.tsx

```diff
@@ -54,6 +55,42 @@ WithErrors.args = {
   color: 'blue',
 };
 
+export const Click: Story<DataCurveProps> = (args) => {
+  const [title, setTitle] = useState('Click on the curve !');
+  const { abscissas, ordinates } = args;
+
+  const abscissaDomain = useDomain(abscissas);
+  const ordinateDomain = useDomain(ordinates);
+  assertDefined(abscissaDomain);
+  assertDefined(ordinateDomain);
+
+  return (
+    <VisCanvas
+      abscissaConfig={{ visDomain: abscissaDomain, showGrid: true }}
+      ordinateConfig={{ visDomain: ordinateDomain, showGrid: true }}
+      title={title}
+    >
+      <DataCurve
+        {...args}
+        onLineClick={(index) => {
+          setTitle(
+            index !== undefined
+              ? `You clicked on segment ${index} going from (${
+                  abscissas[index]
+                }, ${ordinates[index]}) to (${abscissas[index + 1]}, ${
+                  ordinates[index + 1]
+                })`
+              : 'Click on the curve !'
```

Comment:
I would maybe store the index in the state and compute the title in the `VisCanvas` prop above. This will avoid the duplication of the fallback title.